web3c/contract_factory.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @returns a constructor to create a web3.eth.Contract object. The `Contract` object
+ *          here has Oasis specific functionality, such as the ability to specify the
+ *          Oasis contract header upon deploy and the ability to retrieve contract
+ *          expiry from an instance of such a Contract. This is used in both confidential
+ *          and non confidential settings.
+ * @param   {Object} web3 the web3 object.
+ * @param   {Function?} providerFn is a block to execute returning the contract's provider.
+ *          The providerFn should take one argument, the options given upon invoking the
+ *          Contract constructor, for example { saveSession: false } for confidential
+ *          contracts.
+ */
+function make(web3, providerFn) {
```

Comment:
make isn't a meaningful name for this

Reply:
From the caller's perspective, changed from `ContractFactory.make` to `makeContractFactory`.